velox/docs/develop/aggregate-functions.rst

```diff
@@ -92,12 +92,18 @@ Fixed-width part of the accumulator is stored in the row. Variable-width
 part (if exists) is allocated using :doc:`HashStringAllocator <arena>` and a pointer is
 stored in the fixed-width part.
 
-A row is a contiguous byte buffer. Given N aggregates, first N / 8 bytes
-store the null flags, one bit per aggregate, followed by fixed-width
-accumulators.
-
-.. image:: images/aggregation-layout.png
-  :width: 600
+A row is a contiguous byte buffer.  Data is stored in the following order:
+1. Null flags (1 bit per item) for 
+    1. Keys (only if nullable)
+    2. Accumulators
+    3. Dependants
+2. Has-probed flag (1 bit)
+3. Free flag (1 bit)
+4. Keys
+5. Accumulators
+6. Dependants
+7. Variable size (32 bit)
```

Comment:
Can you remind me what this is?

Reply:
It's used to track the variable size of the row (for complex types etc.)